pkg/awx/connection.go

```diff
@@ -128,12 +136,14 @@ func (b *ConnectionBuilder) Build() (c *Connection, err error) {
 	}
 
 	// Check the credentials:
-	if b.username != "" && b.token != "" {
-		err = fmt.Errorf("User name and token are mutually exclusive")
-		return
+	authArgs := 0
+	for _, arg := range [3]string{b.username, b.token, b.bearer} {
+		if len(arg) > 0 {
+			authArgs++
+		}
 	}
-	if b.username == "" && b.token == "" {
-		err = fmt.Errorf("Either user name or token must be provided")
+	if authArgs != 1 {
+		err = fmt.Errorf("Exactly one of user name, token and bearer required")
```

Comment:
`Exactly one of the following is required: username, token or bearer`

Reply:
Done